Collector/ProfileFactory.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+namespace Http\HttplugBundle\Collector;
+
+/**
+ * @author Fabien Bourigault <bourigaultfabien@gmail.com>
+ *
+ * @internal
+ */
+class ProfileFactory
```

Comment:
as stack and profile are basically value objects, it would probably be better to just use the real objects and make assertions on what ends up being in them. this does slightly violate the isolation, but if we still have a isolated unit test on both value objects, it should be good enough. also, we do need the plugin to work correctly with the value objects, so this low level of integration test seems like a good thing to me.